From 11e43fda3486ca41023c429508eccb791684e0a7 Mon Sep 17 00:00:00 2001
From: Durgadoss R <durgadoss.r@intel.com>
Date: Tue, 13 Oct 2015 18:25:08 +0530
Subject: [PATCH] FROM_UPSTREAM [VPG]: drm/i915: Do not use TPS3 for CHV
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

CHV cannot support Training Pattern 3 for Channel EQ part of
link training. Hence, this patch removes support for the same
even when the sink advertises support through DPCD.

Change-Id: I9ab0a3e95b40a987fff4bd6afdffad2f547098b3
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-5971
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: ed63baaf849e91c84ac3e042b1fd6a0af07c16f3
Reviewed-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Signed-off-by: Jani Nikula <jani.nikula@intel.com>
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index a3e5aa3..97e3cb1 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4301,7 +4301,7 @@ intel_dp_get_dpcd(struct intel_dp *intel_dp)
 	}
 
 	/* Training Pattern 3 support */
-	if (intel_dp->dpcd[DP_DPCD_REV] >= 0x12 &&
+	if (!IS_CHERRYVIEW(dev) && intel_dp->dpcd[DP_DPCD_REV] >= 0x12 &&
 	    intel_dp->dpcd[DP_MAX_LANE_COUNT] & DP_TPS3_SUPPORTED) {
 		intel_dp->use_tps3 = true;
 		DRM_DEBUG_KMS("Displayport TPS3 supported");
-- 
1.9.1

